zerver/actions/bots.py

```diff
@@ -40,20 +40,18 @@ def do_change_bot_owner(
                 user_id=user_profile.id,
             ),
         )
-        transaction.on_commit(
-            lambda: send_event(
-                user_profile.realm,
-                delete_event,
-                {previous_owner.id},
-            )
+        send_event(
+            user_profile.realm,
+            delete_event,
+            {previous_owner.id},
         )
```

Comment:
Can you explain the unexpected bug that comes with transactions here? Maybe best posted in #backend for visibility.

Reply:
Started discussion on https://chat.zulip.org/#narrow/stream/3-backend/topic/transaction.2Eon_commit.20when.20sending.20multiple.20events.